mbrl/util/pybullet.py

```diff
@@ -0,0 +1,249 @@
+
+        - ``cfg.overrides.term_fn``: (only for dmcontrol and gym environments) a string
+          indicating the environment's termination function to use when simulating the
+          environment with the model. It should correspond to the name of a function in
+          :mod:`mbrl.env.termination_fns`.
+        - ``cfg.overrides.reward_fn``: (only for dmcontrol and gym environments)
+          a string indicating the environment's reward function to use when simulating the
+          environment with the model. If not present, it will try to use ``cfg.overrides.term_fn``.
+          If that's not present either, it will return a ``None`` reward function.
+          If provided, it should correspond to the name of a function in
+          :mod:`mbrl.env.reward_fns`.
+        - ``cfg.overrides.learned_rewards``: (optional) if present indicates that
+          the reward function will be learned, in which case the method will return
+          a ``None`` reward function.
+        - ``cfg.overrides.trial_length``: (optional) if presents indicates the maximum length
+          of trials. Defaults to 1000.
+
+    Args:
+        cfg (omegaconf.DictConf): the configuration to use.
+
+    Returns:
+        (tuple of env, termination function, reward function): returns the new environment,
+        the termination function to use, and the reward function to use (or ``None`` if
+        ``cfg.learned_rewards == True``).
+    """
+    if "gym___" in cfg.overrides.env:
+        import mbrl.env
+        env = gym.make(cfg.overrides.env.split("___")[1])
+        term_fn = getattr(mbrl.env.termination_fns, cfg.overrides.term_fn)
+        if hasattr(cfg.overrides, "reward_fn") and cfg.overrides.reward_fn is not None:
+            reward_fn = getattr(mbrl.env.reward_fns, cfg.overrides.reward_fn)
+        else:
+            reward_fn = getattr(mbrl.env.reward_fns, cfg.overrides.term_fn, None)
+    else:
+        raise ValueError("Invalid environment string.")
+
+    learned_rewards = cfg.overrides.get("learned_rewards", True)
+    if learned_rewards:
+        reward_fn = None
+
+    if cfg.seed is not None:
+        env.seed(cfg.seed)
+        env.observation_space.seed(cfg.seed + 1)
+        env.action_space.seed(cfg.seed + 2)
+
+    return env, term_fn, reward_fn
+
+def make_env_from_str(env_name: str) -> gym.Env:
+    """Creates a new environment from its string description.
+
+    Args:
+        env_name (str): the string description of the environment. Valid options are:
+          - "gym___<env_name>": a Gym environment (e.g., "gym___HalfCheetah-v2").
+
+    Returns:
+        (gym.Env): the created environment.
+    """
+    if "gym___" in env_name:
+        env = gym.make(env_name.split("___")[1])
+    else:
+        raise ValueError("Invalid environment string.")
+
+    return env
+
+def _is_pybullet_gym_env(env: gym.wrappers.TimeLimit) -> bool:
+    # I don't know how to do this yet
```

Comment:
From a quick look (only checked in one environment), it seems that when you do `gym.make()` a PyBullet environment, it gets wrapped into a `TimeLimit`, and then doing something like `env.env.__module__` should help you identify it's a pybullet environment. For example,

![image](https://user-images.githubusercontent.com/4759586/136027588-d485e565-73c8-4054-bc68-a24d1eca0559.png)

Reply:
See comment here: https://github.com/facebookresearch/mbrl-lib/pull/135#issuecomment-934508729